WASM runtime max heap size

guardrex opened the pull request:
Fixes #30278

Marek ... I'm confused on the setting you provided ... 256 MiB (BTW, they won't allow us to use MiB, so I'm just going to show bytes and write "bytes" for this guidance) ... anyway ... 256 MiB ... or 268,435,456 bytes.

According to https://github.com/dotnet/runtime/blob/main/src/mono/wasm/build/WasmApp.Common.targets, the default is listed a lot higher ...

> - $(EmccMaximumHeapSize) - Maximum heap size specified with `emcc`. Default value: 2147483648 or size of the DLLs, whichever is larger. Corresponds to `-s MAXIMUM_MEMORY=...` emcc arg.

... **2,147,483,648** bytes ... 2 gigs.

So ... is the remark in that file incorrect on the default? If so, what's the actual default value?

Otherwise, the PR is ready for review.

<!-- PREVIEW-TABLE-START -->

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/blazor/host-and-deploy/webassembly.md](https://github.com/dotnet/AspNetCore.Docs/blob/f73b380558e9c31d7fb8cd48aaeb97d8d6092538/aspnetcore/blazor/host-and-deploy/webassembly.md) | [Host and deploy ASP.NET Core Blazor WebAssembly](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/webassembly?branch=pr-en-us-32257) |


<!-- PREVIEW-TABLE-END -->

maraf commented on aspnetcore/blazor/host-and-deploy/webassembly.md:
```diff
@@ -132,6 +132,26 @@ Disable the trimming property if it prevents your app from running normally:
+
+:::moniker range=">= aspnetcore-8.0"
+
+When building a Blazor app that runs on the client (`.Client` project of a Blazor Web App or standalone Blazor WebAssembly app) and targets mobile device browsers, especially Safari on iOS, increasing the maximum memory for the app with the MSBuild property `EmccMaximumHeapSize` might be required. The default value is 2,147,483,648 bytes or size of the app's DLLs, whichever is larger. The followi ...
```
```suggestion
When building a Blazor app that runs on the client (`.Client` project of a Blazor Web App or standalone Blazor WebAssembly app) and targets mobile device browsers, especially Safari on iOS, decreasing the maximum memory for the app with the MSBuild property `EmccMaximumHeapSize` might be required. The default value is 2,147,483,648 bytes or size of the app's DLLs, whichever is larger. The following example sets the value to 268,435,456 bytes in the `Program` file:
```

maraf reviewed (commented):
The default 2GB and we need to decrease it for mobile browsers. They typically don't allow to allocate that much memory, resulting in tab being killed.

guardrex commented:
In that case, we don't need to describe the app's DLLs being larger than the default value. I removed that, and I added a bit on how the default value may too large and prevent the app from running.

guardrex commented on aspnetcore/blazor/host-and-deploy/webassembly.md:
```diff
@@ -132,6 +132,26 @@ Disable the trimming property if it prevents your app from running normally:
+
+:::moniker range=">= aspnetcore-8.0"
+
+When building a Blazor app that runs on the client (`.Client` project of a Blazor Web App or standalone Blazor WebAssembly app) and targets mobile device browsers, especially Safari on iOS, decreasing the maximum memory for the app with the MSBuild property `EmccMaximumHeapSize` may be required. The default value is 2,147,483,648 bytes, which may be too large and result in the app failing to run. ...
```
@maraf ... How about my edit :point_up:.
